Education/Visual Studio features.md

```diff
@@ -0,0 +1,71 @@
+
+- Console App - contains predefined Program.cs class that can be run as console application.
+- Class Library - a base project template for class libraries.
+- ASP.NET Core Web API - contains a structure required for a project to be run as API with mock controller.
+- xUnit Test Project - a base project for xUnit tests.
+- Azure Functions - contains a structure for running a project as Azure function.
+
+Some of these templates provide additional configuration in the setup itself so the code generated is immediately ready to use. For example, Azure Functions project allows you to chose what type of trigger do you want for your function. It also allows you to setup a connection string to the trigger so the generated code can be run and debug locally.
+
+Visual Studio also allows you to create your own templates for the project. The instructions on how to create your custom template can be found on this [link](https://docs.microsoft.com/en-us/visualstudio/ide/how-to-create-project-templates?view=vs-2022).
+
+
+
+### Extensions
+
+VS allows you to add extensions that enhance or alter functionalities provided by the IDE. There are a lot extensions that can be found at 'Extensions' tab or searched on the internet. Some of the extensions are free and some of them are paid. 
+
+Some free extensions that we suggest:
+
+- Fine Code Coverage - provides an overview of your test coverage giving information such as percentage of code covered, risky hotspot, covered branches etc.
+- Format document on Save - automatically applies default code formatting when saving a document.
+- Output enhancer - provides a better visual graphics when checking IDE console logs.
+- Visual Studio Spell Checker - adds spell checking for all files.
+
+
+
+### App settings configuration
+
+Different projects require different setting configurations, like connection string to database or URL to external API etc. These configurations in .NET are usually stored in appsettings.json file. Visual studio allows you to create different configurations through 'Configuration Manager' found in the 'Build' tab. By default projects are run in 'Debug' configuration with appsettings.json but it can be changed to use different settings, example appsettings.Test.json.
+
+Additionally VS allows you to use 'user secrets' by right clicking on project and clicking on 'Manage User Secrets'. These secrets are stored on your machine and cannot be pushed to github. These secrets override appsettings.json and are useful for local development.
+
+
+
+### Connecting to Azure 
+
+VS can be connected to Azure account by clicking on 'Manage connections' in 'Team' tab. When VS is connected to Azure it allows you a several different features, such as accessing DevOps git repository, selecting specific existing trigger (like blob or queue) from your Azure account etc. These Azure connections can also be used to connect to Azure database or other resources and manage them through VS itself.

```

Comment:
Maybe change part `When VS is connected to Azure it allows you a several different features` into `When VS is connected to Azure it allows you to use several different features`

Reply:
Fixed.